### PR TITLE
[signalclirestapi] Add incubator chart signal-cli-rest-api

### PR DESCRIPTION
**Pull request by tamasv**
**Description of the change**

This PR adds a helm chart running Signal messengers REST API. It interacts with Signal messenger service, allowing the users to send and receive p2p encrypted messages using Signal's protocol.

**Benefits**

Currently, there is no other helm chart providing this application, hence it will help Signal users to use their messaging service.

**Checklist** 
- [x] Title of the PR starts with chart name (e.g. `[home-assistant]`)
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Chart `artifacthub.io/changes` changelog annotation has been updated in `Chart.yaml`. See [Artifact Hub documentation](https://artifacthub.io/docs/topics/annotations/helm/#supported-annotations) for more info.
- [x] Variables have been documented in the `values.yaml` file.

### CODE REVIEW

**Review by onedr0p — CHANGES_REQUESTED** [review]

File: charts/incubator/signal-cli-rest-api/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+apiVersion: v2
+appVersion: 1.0.0
```
Comment (onedr0p) [inline charts/incubator/signal-cli-rest-api/Chart.yaml:2]: This should be the app version not the chart version

File: charts/incubator/signal-cli-rest-api/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+description: Dockerized Signal Messenger REST API
+name: signal-cli-rest-api
+version: "0.57"
```
Comment (onedr0p) [inline charts/incubator/signal-cli-rest-api/Chart.yaml:5]: This should be the chart version not the app version

**Review by onedr0p — CHANGES_REQUESTED** [review]

File: charts/incubator/signal-cli-rest-api/Chart.yaml (hunk 2 above)
Comment (tamasv) [reply charts/incubator/signal-cli-rest-api/Chart.yaml]: appversion and version should be fixed now. Thanks!

File: charts/incubator/signal-cli-rest-api/values.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+#
+# This chart inherits from our common library chart. You can check the default values/options here:
+# https://github.com/k8s-at-home/library-charts/tree/main/charts/stable/common/values.yaml
+#
+
+image:
+  # -- image repository
+  repository: bbernhard/signal-cli-rest-api
+  # -- image tag
+  tag: "0.57"
```
[the remark this answers is not in the document]
Comment (tamasv) [reply charts/incubator/signal-cli-rest-api/values.yaml]: Thanks for the hint, image.tag is removed from values.yaml

[the remark this answers is not in the document]
Comment (onedr0p) [reply charts/incubator/signal-cli-rest-api/values.yaml]: You should leave tag but remove the value :)

[the remark this answers is not in the document]
Comment (tamasv) [reply charts/incubator/signal-cli-rest-api/values.yaml]: sorry for that, should be fixed now! :)

**Review by onedr0p — APPROVED** [review]

**Pull request merged**